src/polkit/polkitauthority.c

```diff
@@ -113,9 +114,19 @@ on_proxy_signal (GDBusProxy   *proxy,
                  gpointer      user_data)
 {
   PolkitAuthority *authority = POLKIT_AUTHORITY (user_data);
+  guint16 msg_mask;
+
   if (g_strcmp0 (signal_name, "Changed") == 0)
     {
-      g_signal_emit_by_name (authority, "changed");
+      if ((parameters != NULL) && g_variant_check_format_string(parameters, "(q)", FALSE ) )
+      {
+        g_variant_get(parameters, "(q)", &msg_mask);
+        g_signal_emit (authority, signals[msg_mask], 0);
```

Comment:
Maybe throw in a safety assert that we're not going over `signals` array boundary?

Reply:
Fixed